x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/results/InferenceResults.java

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.inference.results;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+
+public interface InferenceResults<T> extends ToXContentObject, Writeable {
```

Comment:
I would like us to consider an alternative idea here.

Right now this needs to be generic because of the `T value()` method. The paradigm is that we call `value()` to get the result. But how are we going to use this result?

I think eventually the result is an object we append on the object-to-infer, right?

Could we thus have:

`Map<String, Object> result()`? 

It could be we can find a better type than `Map<String, Object>`. However, that would mean that each implementation of the results could be returning an object flexibly. Hard to discuss all this in text so I'm sure it warrants a nice design discussion!

Reply:
> I think eventually the result is an object we append on the object-to-infer, right?

No, the result could be used any number of ways. We don't really append it to the mapped fields of the inference fields, we will supply it to the caller either via an API call, or through the ingest processor (which will have a `target_field` parameter to tell us where to put the result).

> Could we thus have:
`Map<String, Object>` result()?

I would rather not pass around `Map<String, Object>`. If we were going down that path, what is the point of having an object defined at all?